src/test/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/producer/MessageProducerImplUnitTest.java

```diff
@@ -56,23 +62,32 @@
     @Captor
     private ArgumentCaptor<Message> kafkaMessageCaptor;
 
+    @BeforeEach
+    void setup() {
+        when(serializerFactory.getSpecificRecordSerializer(ChipsRestInterfacesSend.class)).thenReturn(avroSerializer);
+    }
+
     @Test
     void testSuccessfulWriteToTopic()
             throws ServiceException, ExecutionException, InterruptedException {
         when(producer.sendAndReturnFuture(any())).thenReturn(mockedFuture);
-
         messageProducerImpl.writeToTopic(getDummyChipsRestInterfacesSend(), TEST_TOPIC);
+
         verify(mockedFuture, times(1)).get();
         verify(producer, times(1)).sendAndReturnFuture(kafkaMessageCaptor.capture());
         Message kafkaMessage = kafkaMessageCaptor.getValue();
+
```

Comment:
i'd also add an extra assert in here, you can tell avroSerializer to return some dummy bytes, then you can assert that the bytes in the captured kafkaMessage are the same that were returned from avroSerializer. This will ensure that the bytes returned from avroSerializer get set correctly in the kafkaMessage sent to the producer.

Reply:
just check it's the same byte[] object

Reply:
Checking this.